easybuild/easyblocks/generic/rubygem.py

```diff
@@ -90,7 +90,7 @@ def install_step(self):
             raise EasyBuildError("Ruby module not loaded?")
 
         # this is the 'proper' way to specify a custom installation prefix: set $GEM_HOME
-        if not self.is_extension:
+        if self.cfg.__dict__.get('software_name') != 'Ruby':
```

Comment:
This is a bit dirty, but I think the real question we should ask is why the existing check based on `self.is_extension` doesn't, it should, see https://github.com/easybuilders/easybuild-framework/blob/master/easybuild/framework/extensioneasyblock.py#L88

Reply:
I think it does not work, because it is a Bundle. So the gems are installed as extension not as an EasyBlock (rubygem). But I might misunderstand the code.

Reply:
@boegel So how to proceed here?

Reply:
OK, I think I see the problem...

We need to have `$GEM_HOME` set both when installing individual Ruby gems as stand-alone modules, and also when installing a bundle of gems together, but not when installing extensions as a part of a `Ruby` installation (i.e. a "batteries included" Ruby installation).

So, I think we need to keep the `not self.is_extension` condition, but combine it with another condition via `or`, like this:

```python
# define $GEM_HOME for both stand-alone installations and bundles of gems,
# to ensure that they are being installed in the right location (i.e. not in parent the Ruby installation)
if not self.is_extension or self.master.name != 'Ruby':
    env.setvar('GEM_HOME', self.installdir)
```

To clarify: `self.master` is made available so extensions can access the parent software (in this case, the bundle that is being installed), see https://github.com/easybuilders/easybuild-framework/blob/master/easybuild/framework/extension.py#L57.

Can you give that a try and see if it works as expected?

Reply:
@hajgato Ping on this?

Reply:
@boegel We have a maintenance period right now, so I will test it when we are online again.